### PR TITLE
Add a cluster saturation time check to the Density test

### PR DESCRIPTION
**Pull request by gmarek**
cc @kubernetes/sig-scalability

### CODE REVIEW

File: test/e2e/density.go [hunk 1]
```diff
@@ -484,6 +494,13 @@ var _ = Describe("Density", func() {
 				name := additionalPodsPrefix + "-" + strconv.Itoa(i)
 				c.Pods(ns).Delete(name, nil)
 			}
+
+			Expect(badEvents).NotTo(BeNumerically(">", int(math.Floor(0.01*float64(totalPods)))))
```
Comment (wojtek-t) [inline test/e2e/density.go]: please remove - it's already done above

File: test/e2e/density.go [hunk 2]
```diff
@@ -484,6 +494,12 @@ var _ = Describe("Density", func() {
 				name := additionalPodsPrefix + "-" + strconv.Itoa(i)
 				c.Pods(ns).Delete(name, nil)
 			}
+
+			saturationThreshold := time.Duration((totalPods / 10)) * time.Second
+			if saturationThreshold < MinSaturationThreshold {
```
Comment (wojtek-t) [inline test/e2e/density.go]: Please make this 10 a constant (minPodThroughput or sth like that)

**Comment by wojtek-t** [comment]
LGTM

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please github issue: #21995

**Comment by timothysc** [comment]
Sorry, fat fingered.  

I was wondering why not report out final throughput as well then?

**Comment by gmarek** [comment]
By final throughput you mean `number of pods / time`?

**Comment by timothysc** [comment]
> By final throughput you mean number of pods / time? 

yup

**Comment by gmarek** [comment]
That I can do. If it'll magically get merged this night I'll create a new PR.

**Comment by wojtek-t** [comment]
LGTM

**Pull request merged**